### PR TITLE
Use multiprocess from pathos for multiprocessing

### PR DESCRIPTION
**Pull request by lhoestq**
[Multiprocess](https://github.com/uqfoundation/multiprocess) (from the [pathos](https://github.com/uqfoundation/pathos) project) allows to use lambda functions in multiprocessed map.
It was suggested to use it by @kandorm.

We're already using dill which is its only dependency.

### CODE REVIEW

File: setup.py
Comment (thomwolf) [inline setup.py:84]: This is install the full pathos package and then we only use the multiprocessing part which override the default multiprocessing package, did I understand it well?

**Comment by lhoestq** [comment]
We can just install multiprocess actually, I'll change that

**Comment by BramVanroy** [comment]
Just an FYI: I remember that I wanted to try pathos a couple of years back and I ran into issues considering cross-platform; the code would just break on Windows. If I can verify this PR by running CPU tests on Windows, let me know!

**Comment by lhoestq** [comment]
That's good to know thanks
I guess we can just wait for #644 to be merged first. I'm working on fixing the tests for windows

**Comment by lhoestq** [comment]
Looks like all the CI jobs on windows passed !
I also tested locally on my windows and it works great :) 

I think this is ready to merge, let me know if you have any remarks @thomwolf @BramVanroy

**Review by thomwolf — APPROVED** [review]
Looks to me!

**Pull request merged**
